qbplatform/quote/models.py

```diff
@@ -20,12 +36,44 @@ class TournamentLocation(models.Model):
     date = models.DateField()
     location = models.CharField(max_length=500, null=True)
 
+    def __str__(self):
+        return self.location + " site of tournament: " + str(self.tournament)
+
 
 class Packet(models.Model):
 
     name = models.CharField(max_length=500)
     tournament = models.ForeignKey(Tournament, on_delete=models.CASCADE)
 
+    def __str__(self):
+        return "Packet " + self.name + " in tournament: " + str(self.tournament)
+
+#   Category model is a hierarchy from top level category (e.g. Literature) to subcategory (e.g. American Literature) 
+class Category(TreeNode):
+
+    name = models.CharField(max_length=100)
+
+    def __str__(self):
+        return self.name
+
+#   Model of a category within a distribution
+#       distinct from Category model so that there exists a notion of a category outside of the confines of a distribution
+#       each (sub)category is allowed a specified range of tossups/bonuses per packet
+#           for example, an editor could specify 1/1 Other Arts but 0-1/0-1 Architecture
+class DistributionCategory(models.Model):
+
+    name = models.CharField(max_length=100)
+
+    min_tossups = models.PositiveSmallIntegerField(null=True)
+    max_tossups = models.PositiveSmallIntegerField(null=True)
+    min_bonuses = models.PositiveSmallIntegerField(null=True)
+    max_bonuses = models.PositiveSmallIntegerField(null=True)
+
+    distribution = models.ForeignKey(Distribution, on_delete=models.CASCADE)
+    categories = models.ManyToManyField(Category)
```

Comment:
This should probably be a `ForeignKey`, right? Otherwise we're saying a subcategory can have more than one parent, which doesn't seem right.